src/components/Tables/Filters.tsx

```diff
@@ -8,25 +8,32 @@ export enum PaginationType {
     Client = "CLIENT",
 }
 
-export type MethodConfig<Data, State> =
-    | {
-          method: (
-              query: PostgrestFilterBuilder<Database["public"], any, any>,
-              state: State
-          ) => PostgrestFilterBuilder<Database["public"], any, any>;
-          paginationType: PaginationType.Server;
-      }
-    | {
-          method: (row: Data, state: State, key: keyof Data) => boolean;
-          paginationType: PaginationType.Client;
-      };
-
-export interface Filter<Data, State> {
+export type ServerSideFilterMethod<DbData extends Record<string, any>, State> = (
+    query: PostgrestFilterBuilder<Database["public"], DbData, any>,
+    state: State
+) => PostgrestFilterBuilder<Database["public"], DbData, any>;
```

Comment:
the actual change that results in the objective of the ticket being satisfied (in so far as the PR description) is passing in DbData as the second type parameter to the PostgrestFilterBuilder type